lib/ex_hash_ring/ring.ex

```diff
@@ -564,8 +564,20 @@ defmodule ExHashRing.Ring do
           {[], 0}
       end
 
-    {:ok,
-     do_find_nodes(table, generation, size, max(num - found_length, 0), hash, found, found_length)}
+    cond do
+      found_overrides_length == num ->
+        {:ok, found_overrides}
+
+      found_overrides_length == 0 ->
+        {:ok, do_find_nodes(table, generation, size, num, hash, [], 0)}
+
+      true ->
+        ring_nodes = do_find_nodes(table, generation, size, num, hash, [], 0)
```

Comment:
[nit] the formatting here is a bit odd, normally a pipeline is begun on the next line indented.

```elixir
ring_nodes = 
  table
  |> do_find_nodes(generation, size, num, hash, [], 0)
  |> Enum.reject(&(&1 in found_overrides)
```

Whether or not to pull `table` out to start the pipeline is debatable, it might be nicer to keep it in the call so there's symmetry between this call and the one on line 572

Reply:
Yeah I think pulling `table` out is less clear. I am spoiled by having pre-commit hooks run `mix format` on my code in the monorepo without needing to do anything